src/components/DevBar/widgets/juju-user.test.tsx

```diff
@@ -0,0 +1,160 @@
+import userEvent from "@testing-library/user-event";
+import { vi } from "vitest";
+
+import * as UserPassForm from "components/LogIn/UserPassForm";
+import * as LocalStorage from "hooks/useLocalStorage";
+import { rootStateFactory } from "testing/factories";
+import { configFactory, generalStateFactory } from "testing/factories/general";
+import { renderComponent } from "testing/utils";
+
+import jujuUser from "./juju-user";
+
+const { Title, Widget } = jujuUser;
+
+function loggedInState() {
+  return rootStateFactory.build({
+    general: generalStateFactory.build({
+      config: configFactory.build({
+        isJuju: true,
+        controllerAPIEndpoint: "wss://jimm.jujucharms.com/api",
+      }),
+      controllerConnections: {
+        "wss://jimm.jujucharms.com/api": {
+          controllerTag: "controller",
+          serverVersion: "1.2.3",
+          user: {
+            "display-name": "eggman",
+            identity: "user-eggman@external",
+            "controller-access": "read",
+            "model-access": "write",
+          },
+        },
+      },
+    }),
+  });
+}
+
+function loggedOutState() {
+  return rootStateFactory.build({
+    general: generalStateFactory.build({
+      config: configFactory.build({
+        isJuju: true,
+        controllerAPIEndpoint: "wss://jimm.jujucharms.com/api",
+      }),
+      controllerConnections: {},
+    }),
+  });
+}
+
+describe("Title", () => {
+  it.for([
+    [true, loggedInState()],
+    [false, loggedOutState()],
+  ] as const)("logged in %s", ([loggedIn, state], { expect }) => {
+    const {
+      result,
+      result: { container },
+    } = renderComponent(<Title />, { state });
+
+    expect(container).toHaveTextContent("Juju User");
+
+    const spans = result.baseElement.getElementsByTagName("span");
+    expect(spans).toHaveLength(1);
+    expect(spans[0]).toHaveClass("show-status");
```

Comment:
Same thing here about getting the element via the text so the order doesn't matter.